hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/PathSegmentCodec.java

```diff
@@ -0,0 +1,275 @@
+                                + str);
+                    }
+                } else if (res.isError()) {
+                    try {
+                        res.throwException();
+                    } catch (CharacterCodingException e) {
+                        throw new IllegalArgumentException("Failed to encode: " + str, e);
+                    }
+                }
+
+                bytes.flip();
+                while (bytes.hasRemaining()) {
+                    byte b = bytes.get();
+                    bld.append('%');
+                    bld.append(asHexDigit((b & 0xF0) >> 4));
+                    bld.append(asHexDigit(b & 0x0F));
+                }
+
+                enc.reset();
+            }
+        }
+
+        return bld.toString();
+    }
+
+    private static CharsetDecoder getDecoder() {
+        CharsetDecoder ret = DECODER.get();
+        if (ret == null) {
+            ret = Charset.forName("UTF-8").newDecoder();
+            DECODER.set(ret);
+        }
+
+        return ret;
+    }
+
+    private static CharsetEncoder getEncoder() {
+        CharsetEncoder ret = ENCODER.get();
+        if (ret == null) {
+            ret = Charset.forName("UTF-8").newEncoder();
+            ENCODER.set(ret);
+        }
+
+        return ret;
+    }
+
+    private static ByteBuffer getByteBuffer() {
+        ByteBuffer ret = BYTE_BUFFER.get();
+        if (ret == null) {
+            ret = ByteBuffer.allocateDirect(6);
+            BYTE_BUFFER.set(ret);
+        }
+
+        return ret;
+    }
+
+    private static CharBuffer getCharBuffer() {
+        CharBuffer ret = CHAR_BUFFER.get();
+        if (ret == null) {
+            ret = CharBuffer.allocate(1);
+            CHAR_BUFFER.set(ret);
+        }
+
+        return ret;
+    }
+
+    private static int asHexDigitValue(char c) {
+        if ('0' <= c && c <= '9') {
+            return c - '0';
+        } else if ('a' <= c && c <= 'f') {
+            return 10 + (c - 'a');
+        } else if ('A' <= c && c <= 'F') {
+            return 10 + (c - 'A');
+        } else {
+            throw new IllegalArgumentException("Invalid hex digit: " + c);
+        }
+    }
+
+    private static char asHexDigit(int b) {
+        if (b < 10) {
+            return (char) ('0' + b);
+        } else {
+            return (char) ('A' + (b - 10));
+        }
+    }
+
+    private static boolean isURISafe(char c) {
+        //made compatible with RestEasy path encoding
+      /*
+       * <a href="http://ietf.org/rfc/rfc3986.txt">RFC 3986</a>
+       *
+       * unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
+       * sub-delims  = "!" / "$" / "&" / "'" / "(" / ")"
+                     / "*" / "+" / "," / ";" / "="
+       * pchar = unreserved / pct-encoded / sub-delims / ":" / "@"
+       *
+       */
+        return ('a' <= c && c <= 'z') || ('A' <= c && c <= 'Z') || ('0' <= c && c <= '9') ||
```

Comment:
Yes, but that's not what I want in my method, because I need to escape any of those characters. So I opted for the "primitive" way of just checking for ranges of chars.